File: backend/app/models/company.py
"""Company workspace document."""
from __future__ import annotations

from pydantic import Field

from app.models.base import MongoModel, PyObjectId
from app.models.enums import ApprovalStatus


class Company(MongoModel):
    name: str
    slug: str
    logo_url: str | None = None
    website: str | None = None
    industry: str | None = None
    size: str | None = None
    about: str | None = None
    locations: list[str] = Field(default_factory=list)
    approval_status: ApprovalStatus = ApprovalStatus.PENDING
    owner_id: PyObjectId
    team: list[PyObjectId] = Field(default_factory=list)

File: backend/src/models/company.js
"use strict";

/** Company workspace document. Mirrors app/models/company.py. */

const mongoose = require("mongoose");
const { Schema } = mongoose;
const { baseSchemaOptions } = require("./plugins");
const { ApprovalStatus, values } = require("./enums");

const companySchema = new Schema(
  {
    name: { type: String, required: true },
    slug: { type: String, required: true, unique: true },
    logo_url: { type: String, default: null },
    website: { type: String, default: null },
    industry: { type: String, default: null },
    size: { type: String, default: null },
    about: { type: String, default: null },
    locations: { type: [String], default: [] },
    linkedin_url: { type: String, default: null },
    hr_email: { type: String, default: null },
    approval_status: {
      type: String,
      enum: values(ApprovalStatus),
      default: ApprovalStatus.PENDING,
    },
    owner_id: { type: String, required: true },
    team: { type: [String], default: [] },
  },
  baseSchemaOptions()
);

companySchema.index({ approval_status: 1 });
companySchema.index({ name: "text", industry: "text" });

const Company = mongoose.model("Company", companySchema, "companies");

module.exports = { Company };
